mutt/regex.c

```diff
@@ -216,52 +218,42 @@ bool mutt_regexlist_match(struct RegexList *rl, const char *str)
  * mutt_regexlist_new - Create a new RegexList
  * @retval ptr New RegexList object
  */
-struct RegexList *mutt_regexlist_new(void)
+struct RegexListNode *mutt_regexlist_new(void)
 {
-  return mutt_mem_calloc(1, sizeof(struct RegexList));
+  return mutt_mem_calloc(1, sizeof(struct RegexListNode));
 }
 
 /**
  * mutt_regexlist_remove - Remove a Regex from a list
- * @param rl  RegexList to alter
+ * @param head  RegexList to alter
  * @param str Pattern to remove from the list
  * @retval 0  Success, pattern was found and removed from the list
  * @retval -1 Error, pattern wasn't found
  *
  * If the pattern is "*", then all the Regexes are removed.
  */
-int mutt_regexlist_remove(struct RegexList **rl, const char *str)
+int mutt_regexlist_remove(struct RegexList *head, const char *str)
 {
-  struct RegexList *p = NULL, *last = NULL;
+  struct RegexListNode *np = STAILQ_FIRST(head), *next = NULL;
   int rc = -1;
 
   if (mutt_str_strcmp("*", str) == 0)
   {
-    mutt_regexlist_free(rl); /* "unCMD *" means delete all current entries */
+    mutt_regexlist_free(head); /* "unCMD *" means delete all current entries */
     rc = 0;
   }
   else
   {
-    p = *rl;
-    last = NULL;
-    while (p)
+    /* I simplized code with STAILQ, i am not sure here too, please corect me*/
```

Comment:
Rather than commenting the code, you can push to GitHub, then add questions/comments here in the issue.
That way, if the code **doesn't** need changing, we don't need another commit just to remove the comment.